Change 27203 by rgs@stencil on 2006/02/16 14:15:53

	Subject: Problem compiling swigged c++ code with 5.8.8
	From: Merijn Broeren <merijnb@iloquent.com>
	Date: Wed, 15 Feb 2006 13:51:49 +0100
	Message-ID: <20060215125148.GA12535@brugman.iloquent.nl>

Affected files ...

... //depot/perl/XSUB.h#103 edit
... //depot/perl/perl.h#665 edit

Differences ...

--- perl.h	2006/02/12 00:10:22	19541
+++ perl.h	2006/02/16 22:15:53	19572
@@ -211,7 +211,7 @@
 #endif
 
 #ifndef PERL_UNUSED_DECL
-#  ifdef HASATTRIBUTE_UNUSED
+#  if defined(HASATTRIBUTE_UNUSED) && !defined(__cplusplus)
 #    define PERL_UNUSED_DECL __attribute__unused__
 #  else
 #    define PERL_UNUSED_DECL
--- XSUB.h	2006/02/02 22:24:03	19447
+++ XSUB.h	2006/02/16 22:15:53	19572
@@ -91,7 +91,7 @@
 #if defined(__CYGWIN__) && defined(USE_DYNAMIC_LOADING)
 #  define XS(name) __declspec(dllexport) void name(pTHX_ CV* cv)
 #else
-#  ifdef HASATTRIBUTE_UNUSED
+#  if defined(HASATTRIBUTE_UNUSED) && !defined(__cplusplus)
 #    define XS(name) void name(pTHX_ CV* cv __attribute__unused__)
 #  else
 #    define XS(name) void name(pTHX_ CV* cv)
